File: lapi/internal/modules/auth/types/auth_types.go
package types

import "github.com/google/uuid"

// NoBody is used for tonic handlers that take no request body.
type NoBody struct{}

// ── Path params ─────────────────────────────────────────────────────

type AddressIDPath struct {
	AddressID uuid.UUID `path:"id"`
}

// ── Profile ─────────────────────────────────────────────────────────

type UpdateProfileRequest struct {
	FirstName *string `json:"first_name,omitempty" validate:"omitempty,min=2,max=100"`
	LastName  *string `json:"last_name,omitempty" validate:"omitempty,min=2,max=100"`
	Phone     *string `json:"phone,omitempty" validate:"omitempty,max=20"`
}

type ProfileResponse struct {
	ID             uuid.UUID         `json:"id"`
	KeycloakUserID string            `json:"keycloak_user_id,omitempty"`
	Email          string            `json:"email"`
	FirstName      string            `json:"first_name"`
	LastName       string            `json:"last_name"`
	Phone          string            `json:"phone"`
	Role           string            `json:"role"`
	Addresses      []AddressResponse `json:"addresses"`
}

// ── Address ─────────────────────────────────────────────────────────

type CreateAddressRequest struct {
	Label      string `json:"label" validate:"max=50"`
	Street     string `json:"street" validate:"required,max=255"`
	City       string `json:"city" validate:"required,max=100"`
	PostalCode string `json:"postal_code" validate:"required,max=10"`
	IsDefault  bool   `json:"is_default"`
}

// UpdateAddressInput combines path param + body for tonic binding.
type UpdateAddressInput struct {
	AddressIDPath
	UpdateAddressRequest
}

type UpdateAddressRequest struct {
	Label      *string `json:"label,omitempty" validate:"omitempty,max=50"`
	Street     *string `json:"street,omitempty" validate:"omitempty,max=255"`
	City       *string `json:"city,omitempty" validate:"omitempty,max=100"`
	PostalCode *string `json:"postal_code,omitempty" validate:"omitempty,max=10"`
	IsDefault  *bool   `json:"is_default,omitempty"`
}

// DeleteAddressInput is path-only for tonic binding.
type DeleteAddressInput struct {
	AddressIDPath
}

type AddressResponse struct {
	ID         uuid.UUID `json:"id"`
	Label      string    `json:"label"`
	Street     string    `json:"street"`
	City       string    `json:"city"`
	PostalCode string    `json:"postal_code"`
	IsDefault  bool      `json:"is_default"`
}

type ErrorResponse struct {
	Error   string `json:"error"`
	Message string `json:"message"`
}

File: lapiweb/types/authTypes.ts
// Maps to: identity/types + auth/types Go DTOs

// ── Identity (Keycloak) ─────────────────────────────────────────────

export interface RegisterRequest {
  email: string;
  password: string;
  full_name: string;
  phone_number?: string;
}

export interface RegisterResponse {
  keycloak_user_id: string;
  email: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  access_token: string;
  refresh_token: string;
  expires_in: number;
}

export interface RefreshTokenRequest {
  refresh_token: string;
}

export interface ForgotPasswordRequest {
  email: string;
}

export interface ChangePasswordRequest {
  new_password: string;
}

export interface SocialLoginURLResponse {
  url: string;
  provider: string;
}

export interface SocialCallbackRequest {
  code: string;
  state?: string;
  redirect_uri: string;
}

// ── Profile ─────────────────────────────────────────────────────────

export interface ProfileResponse {
  id: string;
  keycloak_user_id?: string;
  email: string;
  first_name: string;
  last_name: string;
  phone: string;
  role: "client" | "admin";
  addresses: AddressResponse[];
}

export interface UpdateProfileRequest {
  first_name?: string;
  last_name?: string;
  phone?: string;
}

// ── Address ─────────────────────────────────────────────────────────

export interface AddressResponse {
  id: string;
  label: string;
  street: string;
  city: string;
  postal_code: string;
  is_default: boolean;
}

export interface CreateAddressRequest {
  label?: string;
  street: string;
  city: string;
  postal_code: string;
  is_default?: boolean;
}

export interface UpdateAddressRequest {
  label?: string;
  street?: string;
  city?: string;
  postal_code?: string;
  is_default?: boolean;
}
